### PR TITLE
2.0: Refactor code and add documentation

### PR DESCRIPTION
**Pull request by andrewzah**
Hi, I was working on 한글 manipulation code for my project and then I found this library.

I updated and refactored the code to be more idiomatic rust code.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,359 +1,373 @@
...
+/// Checks if a char is a middle/vowel jamo, or jungseong (중성).
+///
+/// ```rust
+/// use hangeul::is_jungseong; //is_vowel
+///
+/// assert_eq!(false, is_jungseong('ㅉ' as u32));
+/// assert_eq!(true, is_jungseong('ㅏ' as u32));
+/// assert_eq!(false, is_jungseong('ㄵ' as u32));
+/// ```
+pub fn is_jungseong(code: u32) -> bool {
```
Comment (bekker) [inline src/lib.rs:246]: Is this function necessary? It's completely same as `is_moeum`. Or use aliases, maybe?

File: src/lib.rs [hunk 2]
```diff
@@ -1,359 +1,373 @@
...
+const COMPAT_JUNGSEONG_START: u32 = 0x314F;
+const COMPAT_JUNGSEONG_END: u32 = 0x3163;
+
+// tail/bottom/end chars
+const JONGSEONG_START: u32 = 0x11A8;
+const JONGSEONG_END: u32 = 0x11C2;
+const COMPAT_JONGSEONG_START: u32 = 0x3165;
+const COMPAT_JONGSEONG_END: u32 = 0x318E;
+
+/// Check if the syllable is a finished/composed Hangeul syllable.
```
Comment (bekker) [inline src/lib.rs]: `Check if the syllable` should be `Check if a u32`

**Review by bekker — COMMENTED** [review]
Thank you for the great work :)

File: src/lib.rs (hunk 1 above)
Comment (andrewzah) [reply src/lib.rs:246]: :+1: I forgot to update the code to `Jungseong::from_u32(code).is_some()`.

**Pull request merged**